webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -181,85 +178,62 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     }
   }, [settings.filterset, isLoadingSettings]);
 
-  const [selectedExperimentIds, setSelectedExperimentIds] = useState<number[]>(() => {
-    if (isLoadingSettings) {
-      return [];
-    }
-    return settings.selectedExperiments;
-  });
-  const [excludedExperimentIds, setExcludedExperimentIds] = useState<Set<number>>(() => {
-    if (isLoadingSettings) {
-      return new Set();
-    }
-    return new Set(settings.excludedExperiments);
-  });
   const [isLoading, setIsLoading] = useState(true);
   const [error] = useState(false);
   const [canceler] = useState(new AbortController());
 
-  const colorMap = useGlasbey(selectedExperimentIds);
+  const colorMap = useGlasbey(settings.selectedExperiments);
   const { height: containerHeight, width: containerWidth } = useResize(contentRef);
   const height =
     containerHeight - 2 * parseInt(getCssVar('--theme-stroke-width')) - (isPagedView ? 40 : 0);
   const [scrollPositionSetCount] = useState(observable(0));
 
-  useMemo(() => {
-    if (isLoading) {
-      return;
+  const selectedExperimentIds: Set<number> = useMemo(() => {
+    return isLoadingSettings ? new Set() : new Set(settings.selectedExperiments);
+  }, [isLoadingSettings, settings.selectedExperiments]);
+
+  const excludedExperimentIds: Set<number> = useMemo(() => {
+    return isLoadingSettings ? new Set() : new Set(settings.excludedExperiments);
+  }, [isLoadingSettings, settings.excludedExperiments]);
```

Comment:
I'm wondering if there's a better approach to how we're dealing with selection here:

1) grid selection is managed separately from the id selection, but should be linked, so we either have to update one when the other updates (like in handleSelectionChange) or handle it as an effect.

2) id selection has two lists -- which one is used is based on a boolean, all of which are managed separately.

maybe we should instead have a single object representing the id selection including the select all state and derive the grid selection from that as a memo?

something like
```tsx
const IDSelection = {
  type: 'exclusive' | 'inclusive', // exclusive is select all, inclusive is not
  ids: string[]
}
```

Reply:
I had the same thought and started down that path, and decided to hold off on that task. There are some minor nuances between the two sets of ids that differed enough where it wasn't a straight forward refactor.